src/app/FakeLib/SemVerHelper.fs

```diff
@@ -76,6 +76,15 @@ type SemVerInfo =
                     0
             | _ -> invalidArg "yobj" "cannot compare values of different types"
 
+
+let private SemVerPattern = "^(?:0|[1-9][0-9]*)\.(?:0|[1-9][0-9]*)\.(?:0|[1-9][0-9]*)(?:-[\da-zA-Z\-]+(?:\.[\da-zA-Z\-]+)*)?(?:\+[\da-zA-Z\-]+(?:\.[\da-zA-Z\-]+)*)?$"
```

Comment:
Ok let's ask differently. Why do you need this new function? What's the use
case?
On May 25, 2015 4:40 PM, "Nathanael Jones" <notifications@github.com> wrote:

> In src/app/FakeLib/SemVerHelper.fs
> <https://github.com/fsharp/FAKE/pull/811#discussion_r30982897>:
>
> > @@ -76,6 +76,15 @@ type SemVerInfo =
> >                      0
> >              | _ -> invalidArg "yobj" "cannot compare values of different types"
> >
> > +
> > +let private SemVerPattern = "^(?:0|[1-9][0-9]*)\.(?:0|[1-9][0-9]*)\.(?:0|[1-9][0-9]*)(?:-[\da-zA-Z\-]+(?:\.[\da-zA-Z\-]+)*)?(?:\+[\da-zA-Z\-]+(?:\.[\da-zA-Z\-]+)*)?$"
>
> You're saying we should throw an exception during parse for parsable but
> technically invalid semver strings? Int32.Parse doesn't align with the
> spec, particularly if we think about culture. While we can improve parse
> (and possibly make it more strict), some things (like prohibiting leading
> zeroes) are more complex without a regex.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fsharp/FAKE/pull/811/files#r30982897>.
>